switch_theme.cgi

```diff
@@ -0,0 +1,47 @@
+#!/usr/local/bin/perl
+# switch_theme.cgi
+# Change the theme for the current user if allowed
+
+BEGIN { push(@INC, "."); };
+use WebminCore;
+
+&init_config();
+&ReadParse();
+my $err = sub {
+    &PrintHeader();
+    print("<tt>Cannot change theme : $_[0]</tt>\n");
+    exit(1);
+    };
+# Check if in debug mode
+&$err("Debug mode is not enabled!")
+    if (!$gconfig{'error_stack'} &&
+        !$gconfig{'debug_enabled'} &&
+        !$gconfig{'debug_theme_switcher'});
+# Check if allowed to change theme,
+# otherwise throw an error
+if (!&foreign_available('theme') &&
+    !&foreign_available('change-user') &&
+    !&foreign_available('webmin') &&
+    !&foreign_available('acl')) {
+    &$err("You are not allowed to change themes!"); 
+    }
+# Check if the theme is known
+&$err("Theme identification is not known!")
+    if (!$in{'theme'} || $in{'theme'} !~ /^[123]$/);
+# Check if the remote user is known
+&$err("Remote user is not known!") if (!$remote_user);
+# Define the theme
+my $themes = {
+    '1' => 'authentic-theme',
+    '2' => 'gray-theme',
+    '3' => '' };
+my $theme = $themes->{$in{'theme'}};
+# Change the theme
```

Comment:
Better to use the acl module's modify_user API for this, as it will work with other backends like LDAP

Reply:
We don't modify a user. We modify local copies of Webmin `config` and `miniserv.conf` files.

Reply:
Yes, but the `acl::modify_user` function will also update those files using an API that will work even if we change the underlying format for storing the theme setting.

Reply:
> Yes, but the `acl::modify_user` function will also update those files using an API that will work even if we change the underlying format for storing the theme setting.

But can theme parameter used by Webmin be stored somewhere else? What would be the practical purpose of using that sub, if we only need to update the local version of `config` and `miniserv.conf` files? Unless your suggestion is more of a theoretical one?

Reply:
Yes, we support storing users in LDAP, MySQL or PostgreSQL. Only the acl::modify_user function will update the correct data store for the user, which might not be miniserv.conf

Reply:
> Yes, we support storing users in LDAP, MySQL or PostgreSQL. Only the acl::modify_user function will update the correct data store for the user, which might not be miniserv.conf

Yes, but does this data include `preroot_` from `miniserv.conf` file?

Reply:
Yes when a user is in LDAP or MySQL, there is no preroot_ line in miniserv.conf

Reply:
> Yes when a user is in LDAP or MySQL, there is no preroot_ line in miniserv.conf

Cool! Thanks!

I have changed the code to use ACL `modify_user` insted!